TypeCobol/Compiler/CupPreprocessor/CompilerDirectiveBuilder.cs

```diff
@@ -172,13 +183,42 @@ public virtual void EnterCopyCompilerStatement(TypeCobol.Compiler.Scanner.Token
             }
         }
 
-        private static void BuildReplaceOperation(IList<ReplaceOperation> replaceOperations, ref Token comparisonToken, ref Token[] followingComparisonTokens, ref Token replacementToken, ref Token[] replacementTokens, bool replaceTokens, List<Token> operandTokens)
+        /// <summary>
+        /// Flat a list by extracting tokens in a GroupToken
+        /// </summary>
+        /// <param name="list">The list to be flattened</param>
+        /// <returns>The Flattened list</returns>
+        private List<Token> FlatList(List<Token> list)
+        {
+            if (list == null || list.Count == 0)
+                return list;
+            if(list.Exists(t => t is GroupToken))
+            {
+                this.CompilerDirective.HasGroupToken = true;
+                List<Token> flatList = new List<Token>();
+                foreach(Token t in list)
+                {
+                    if (t is GroupToken gt)
+                    {
+                        flatList.AddRange(gt.Group);
+                    }
+                    else
+                    {
+                        flatList.Add(t);
+                    }                   
+                }
+                return flatList;
+            }
+            return list;
+        }
+        private bool BuildReplaceOperation(IList<ReplaceOperation> replaceOperations, ref Token comparisonToken, ref Token[] followingComparisonTokens, ref Token replacementToken, ref Token[] replacementTokens, bool replaceTokens, List<Token> operandTokens)
```

Comment:
The ANTLR-related duplicate method in `TypeCobol.Compiler.Preprocessor.CompilerDirectiveBuilder` has not been updated.

Reply:
I think we can remove the Antlr Preprocessor.
I've created #1788 about that.